inst/include/RcppSimdJson/deserialize/scalar.hpp

```diff
@@ -35,85 +35,85 @@ inline auto get_scalar(simdjson::dom::element element) noexcept(noxcpt<R_Type>()
 template <>
 inline auto
 get_scalar_<bool, rcpp_T::chr>(simdjson::dom::element element) noexcept(noxcpt<rcpp_T::chr>()) {
-    return element.get<bool>().first ? Rcpp::String("TRUE") : Rcpp::String("FALSE");
+    return bool(element) ? Rcpp::String("TRUE") : Rcpp::String("FALSE");
```

Comment:
All of these changes in scalar.hpp are based on casting. As you can see, it makes the code really nice. A cost is a throwing operation in simdjson: if the cast is not possible, we throw. These functions are marked `noexcept(noxcpt<rcpp_T::chr>())` which might mean that they are not supposed to throw. Well, if they do, the exception won't propagate.

Note that the current code would be evil in case of error. It would not throw but it would not do anything good.